kimchi/src/proof.rs

```diff
@@ -78,12 +105,15 @@ pub struct ProverProof<G: AffineCurve> {
     pub evals: [ProofEvaluations<Vec<ScalarField<G>>>; 2],
 
     /// Required evaluation for [Maller's optimization](https://o1-labs.github.io/mina-book/crypto/plonk/maller_15.html#the-evaluation-of-l)
+    #[serde_as(as = "o1_utils::serialization::SerdeAs")]
     pub ft_eval1: ScalarField<G>,
 
     /// The public input
+    #[serde_as(as = "Vec<o1_utils::serialization::SerdeAs>")]
     pub public: Vec<ScalarField<G>>,
 
     /// The challenges underlying the optional polynomials folded into the proof
+    #[serde_as(as = "Vec<(Vec<o1_utils::serialization::SerdeAs>, serde_with::Same)>")]
```

Comment:
Oh, `serde_with::Same` 👍🏻 I tried to make the tuple work for serialization but didn't find any way to make it compile, reason why I created the `RecursionChallenge` type. Then I had to change the ocaml types, both here and in the mina side. I still find this particular type more readable than a tuple, but it is great to learn a way to make it work without a special type to handle it. Perhaps in the future we may want to include a struct here?